Makefile

```diff
@@ -473,11 +473,21 @@ govet: ## Run govet on Go source files in the repository.
     ./test/k8s/... \
     ./tools/...
 
-lint: ## Run golangci-lint and check if the helper headers in bpf/mock are up-to-date.
+golangci-lint: ## Run golangci-lint
+ifneq (,$(findstring $(GOLANGCILINT_WANT_VERSION),$(GOLANGCILINT_VERSION)))
+golangci-lint:
 	@$(ECHO_CHECK) golangci-lint
 	$(QUIET) golangci-lint run
+else
+golangci-lint:
+	$(QUIET) $(CONTAINER_ENGINE) run --rm -v `pwd`:/app -w /app docker.io/golangci/golangci-lint:v$(GOLANGCILINT_WANT_VERSION) golangci-lint run
+endif
+
+bpf-mock-check: ## Check if the helper headers in bpf/mock are up-to-date.
 	$(QUIET) $(MAKE) $(SUBMAKEOPTS) -C bpf/mock/ check_helper_headers
 
+lint: golangci-lint bpf-mock-check
```

Comment:
Can we keep some documentation for the `lint` target? I think the comment after the `##` is parsed and displayed for `make help`.